src/sentry/ingest/transaction_clusterer/rules.py

```diff
@@ -53,11 +53,19 @@ def write(self, project: Project, rules: RuleSet) -> None:
 class ProjectOptionRuleStore:
     _option_name = "sentry:transaction_name_cluster_rules"
 
+    def read_sorted(self, project: Project) -> List[Tuple[ReplacementRule, int]]:
+        return project.get_option(self._option_name, default=[])  # type: ignore
+
     def read(self, project: Project) -> RuleSet:
-        return project.get_option(self._option_name, default={})  # type: ignore
+        return {rule: last_seen for rule, last_seen in self.read_sorted(project)}
+
+    def _sort(self, rules: RuleSet) -> List[Tuple[ReplacementRule, int]]:
+        """Sort rules by number of slashes, i.e. depth of the rule"""
+        return sorted(rules.items(), key=lambda p: p[0].count("/"), reverse=True)
```

Comment:
> `lambda p: p[0].count("/")`

I guess this is good enough for now, but we should be aware that we may need to improve this in the future. Transactions ending with `/` have an extra length point than what they should have, and we're also assuming there are no double slashes.

Reply:
Agreed.